include/ensmallen_bits/lookahead/lookahead.hpp

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file lookahead.hpp
+ * @author Marcus Edel
+ *
+ * Lookahead is a stochastic gradient based optimization method which chooses a
+ * search direction by looking ahead at the sequence of "fast weights" generated
+ * by another optimizer.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_LOOKAHEAD_LOOKAHEAD_HPP
+#define ENSMALLEN_LOOKAHEAD_LOOKAHEAD_HPP
+
+#include <ensmallen_bits/sgd/decay_policies/no_decay.hpp>
+
+namespace ens {
+
+/**
+ * Lookahead is a stochastic gradient based optimization method which chooses a
+ * search direction by looking ahead at the sequence of "fast weights" generated
+ * by another optimizer.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @article{Zhang2019,
+ *   author  = {Michael R. Zhang and James Lucas and Geoffrey E. Hinton and
+ *              Jimmy Ba},
+ *   title   = {Lookahead Optimizer: k steps forward, 1 step back},
+ *   journal = {CoRR},
+ *   year    = {2019},
+ *   url     = {http://arxiv.org/abs/1907.08610}
+ * }
```

Comment:
I think we need an `@endcode` here too.

Reply:
Nice catch, looks like there are a couple more optimizer where the `@endcode` or `@code` is missing, so I will fix those as well.